tests/unit/src/test/scala/tests/ClassNameResolverSuite.scala

```diff
@@ -0,0 +1,203 @@
+package tests
+
+import java.nio.file.Paths
+
+import scala.meta.inputs.Input
+import scala.meta.internal.metals.Buffers
+import scala.meta.internal.metals.BuildTargets
+import scala.meta.internal.metals.ScalaVersionSelector
+import scala.meta.internal.metals.UserConfiguration
+import scala.meta.internal.metals.{BuildInfo => V}
+import scala.meta.internal.mtags.MtagsEnrichments._
+import scala.meta.internal.parsing.ClassFinder
+import scala.meta.internal.parsing.Trees
+import scala.meta.io.AbsolutePath
+
+import munit.FunSuite
+import munit.TestOptions
+
+class ClassNameResolverSuite extends FunSuite {
```

Comment:
@tgodzik @ckipp01 what do you think about how class name resolving works? It's cursor aware, but if the cursor isn't pointing to `Defn`, then it tries to fallback to the first defined class/object/definition, etc.

Reply:
I don't think we should have any kind of fallback here, there should be an error in that case unless there is only single class in the file. It will be suprising to fallback to the first one, why not the closest one or maybe the one that is named the same as the file, or the last one? There is no 100% safe fallback here.

Reply:
Yea I would also find it maybe a bit odd to fallback to the first one. I think I like the idea of the closest if that's possible.

Reply:
Together with @tgodzik, we came to the following solution: if there is any ambiguity about which class/def metals can show, we should present all options to the user and let him pick one of them. I think it's a great idea and an easy solution to this problem